visualization_msgs/msg/Marker.msg

```diff
@@ -56,6 +56,16 @@ geometry_msgs/Point[] points
 # NOTE: alpha is not yet used
 std_msgs/ColorRGBA[] colors
 
+# Only used if the marker has a UV mapping defined for it or one is defined using uv_coordinates.
+# A URI to where the texture is accessible.
+string texture_map
+# Only used when a texture_map is defined and the type specified has some use.
+# Location of each vertex within the texture map; in the range: [0.0-1.0]
+UVCoordinate[] uv_coordinates
+# Only used when a texture_map is defined.
+# Causes the rendering engine to reload the given texture if already loaded.
+bool texture_update
```

Comment:
Maybe the texture could be reloaded every time a marker with the texture is added? If not, does one marker's `texture_update=True` cause all markers using the same texture to get updated when the texture updates?

Reply:
I'm just going to answer this question based on my implementation (which can change depending on preference).

To answer @sloretz,
I currently load textures under their URI into Ogre. This means that if two different markers use the same URI, they will pull the same loaded texture. This also means if one updates, they will both now be using the same updated texture. This can change by appending some metadata to the name used to cache the texture in Ogre such that updating that specific marker without touching others.

To answer @tfoote,
The textures can be of arbitrary size. I have no idea how large people might want to make textures but I think that we can expect up to 4k images (which may be an unreasonable assumption). I imagine for really large surfaces, people will likely just use meshes instead of markers so maybe we can make an assertion about performance loss when textures exceed a certain size and have them be included in the message in their raw format. I was using a URI and caching it in Ogre to minimize data transmission assuming people might want to use very large images for their textures (no idea what the application might be :shrug:).

With regards to modification,
The reason why you would want a separate "modify" param is that the texture may be quite large and you want that to be a one-time cost. Reloading every time a marker with the texture is added could be a way to handle it but I think it's scenario dependent so it's really up to if we want to make that decision for them or let them decide for themselves with the `texture_update` param.

Reply:
Removed texture update following the decision to change to texture in message.